cardano-api/src/Cardano/Api/Orphans.hs

```diff
@@ -0,0 +1,225 @@
+           , "txIndex" .= txIndex
+           , "certIndex" .= certIndex
+           ]
+
+
+instance Crypto.Crypto crypto => ToJSON (Shelley.PState crypto) where
+  toJSON pState = object [ "currentParams" .= Shelley._pParams pState
+                         , "futureParams" .= Shelley._fPParams pState
+                         , "retiringPools" .= Shelley._retiring pState
+                         ]
+
+instance ( Consensus.ShelleyBasedEra era
+         , ToJSON (Core.TxOut era)
+         ) => ToJSON (Shelley.UTxO era) where
+  toJSON (Shelley.UTxO utxo) = toJSON utxo
+
+instance ( Consensus.ShelleyBasedEra era
+         , ToJSON (Core.Value era)
+         ) => ToJSON (Shelley.TxOut era) where
+  toJSON (Shelley.TxOut addr amount) =
+    object
+      [ "address" .= addr
+      , "amount" .= amount
+      ]
+
+instance Crypto.Crypto crypto => ToJSON (Shelley.TxIn crypto) where
+  toJSON = toJSON . txInToText
+
+instance Crypto.Crypto crypto => ToJSONKey (Shelley.TxIn crypto) where
+  toJSONKey = toJSONKeyText txInToText
+
+txInToText :: Crypto.Crypto crypto => Shelley.TxIn crypto -> Text
+txInToText (Shelley.TxIn (Shelley.TxId txidHash) ix) =
+  hashToText txidHash
+    <> Text.pack "#"
+    <> Text.pack (show ix)
+
+hashToText :: Crypto.Hash crypto a -> Text
+hashToText = Text.decodeLatin1 . Crypto.hashToBytesAsHex
+
+instance Crypto.Crypto crypto => ToJSON (Shelley.NonMyopic crypto) where
+  toJSON nonMy = object [ "nonMyopicLikelihoods" .= Shelley.likelihoodsNM nonMy
+                        , "rewardPot" .= Shelley.rewardPotNM nonMy
+                        ]
+
+instance ToJSON Shelley.Likelihood where
+  toJSON (Shelley.Likelihood llhd) =
+    toJSON $ fmap (\(Shelley.LogWeight f) -> exp $ realToFrac f :: Double) llhd
+
+instance Crypto.Crypto crypto => ToJSON (Shelley.SnapShots crypto) where
+  toJSON ss = object [ "pstakeMark" .= Shelley._pstakeMark ss
+                     , "pstakeSet" .= Shelley._pstakeSet ss
+                     , "pstakeGo" .= Shelley._pstakeGo ss
+                     , "feeSnapShot" .= Shelley._feeSS ss
+                     ]
+
+instance Crypto.Crypto crypto => ToJSON (Shelley.SnapShot crypto) where
+  toJSON ss = object [ "stakeSnapShot" .= Shelley._stake ss
+                     , "delegationSnapShot" .= ShelleyEpoch._delegations ss
+                     , "poolParamsSnapShot" .= Shelley._poolParams ss
+                     ]
+
+instance ToJSON (Shelley.Stake crypto) where
+  toJSON (Shelley.Stake s) = toJSON s
+
+instance Crypto.Crypto crypto => ToJSON (Shelley.RewardUpdate crypto) where
+  toJSON rUpdate = object [ "rewUpdateDeltaT" .= Shelley.deltaT rUpdate
+                          , "rewUpdateDeltaR" .= Shelley.deltaR rUpdate
+                          , "rewUpdateRs" .= Shelley.rs rUpdate
+                          , "rewUpdateDeltaF" .= Shelley.deltaF rUpdate
+                          , "rewUpdateNonMyopic" .= Shelley.nonMyopic rUpdate
+                          ]
```

Comment:
For this and other types, I'm thinking we shouldn't qualify field names this way.  For example just `deltaT` seems find.

Reply:
:+1: Making the change